osu.Game/Online/Rooms/Room.cs

```diff
@@ -348,7 +348,7 @@ public Room()
 
         public Room(MultiplayerRoom room)
         {
-            RoomID = room.RoomID;
+            RoomID = room.RoomID > 0 ? room.RoomID : null;
```

Comment:
See explanation in https://github.com/ppy/osu/pull/32621/commits/67eb691c8310620b629211626ec5425de9b2538f

Reply:
Did you explore alternative options to this, like incrementing room IDs on the `MultiplayerRoom` instead?

I don't necessarily think there's any real chance this will go wrong in its current form, at least not soon (the database row is `UNSIGNED BIGINT`, which is good, but... we're also using `long` and not `ulong` for the ID, so overflow shenanigans *may* occur...)

I am however somewhat wary of playing these sorts of ID autoconversion games in model classes, because when they go bad, they usually go _catastrophic_.

Reply:
Okay. Aside from the whole typing issue because it's not really relevant to this PR, let's try another path which always increments that ID https://github.com/ppy/osu/pull/32621/commits/74d234a7debd6be49a8d6eba8045368ab65d1b07

This isn't a multiplayer issue, actually. The code was originally there for playlists I believe, though I can't find any broken tests aside from daily challenge (which were easy to resolve).

Hopefully that makes it more palatable?